src/app/auth.service.ts

```diff
@@ -0,0 +1,46 @@
+import { UserService } from './user.service';
+import { AppUser } from './models/app-user';
+import { Injectable } from '@angular/core';
+import { AngularFireAuth } from '@angular/fire/auth';
+import * as firebase from 'firebase';
+import { ActivatedRoute } from '@angular/router';
+import { switchMap } from 'rxjs/operators';
+import { Observable, of } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  user$: Observable<firebase.User>;
```

Comment:
i know in the tutorial he uses this syntax for observables but let's drop it for readability. instead of user$ to denote an observable lets use userObservsable